vote/src/vote_account.rs

```diff
@@ -185,15 +219,25 @@ impl VoteAccounts {
         if stake == 0u64 {
             return;
         }
+
         let Some(staked_nodes) = self.staked_nodes.get_mut() else {
             return;
         };
-        if let Some(node_pubkey) = vote_account.node_pubkey() {
+
+        VoteAccounts::do_add_node_stake(staked_nodes, stake, vote_account.node_pubkey().copied());
+    }
+
+    fn do_add_node_stake(
+        staked_nodes: &mut Arc<HashMap<Pubkey, u64>>,
+        stake: u64,
+        node_pubkey: Option<Pubkey>,
+    ) {
+        node_pubkey.map(|node_pubkey| {
             Arc::make_mut(staked_nodes)
                 .entry(node_pubkey)
```

Comment:
```suggestion
    fn do_add_node_stake(
        staked_nodes: &mut Arc<HashMap<Pubkey, u64>>,
        stake: u64,
        node_pubkey: Option<&Pubkey>,
    ) {
        node_pubkey.map(|node_pubkey| {
            Arc::make_mut(staked_nodes)
                .entry(node_pubkey.clone())
```
Might as well only allocate if node_pubkey.is_some()?

Reply:
nice catch!